tests/integration.rs

```diff
@@ -460,6 +464,97 @@ fn fw_path(_fw_type: FwType) -> String {
     fw_path.to_str().unwrap().to_string()
 }
 
+struct MetaEvent {
+    event: String,
+    device_id: Option<String>,
+}
+
+impl MetaEvent {
+    pub fn match_with_json_event(&self, v: &serde_json::Value) -> bool {
+        let mut matched = false;
+        if v["event"].as_str().unwrap() == self.event {
+            if let Some(device_id) = &self.device_id {
+                if v["properties"]["id"].as_str().unwrap() == device_id {
+                    matched = true
+                }
+            } else {
+                matched = true;
+            }
+        }
+        matched
+    }
+}
+
+// Parse the event_monitor file based on the format that each event
+// is surrounded by '{' and '}'
+fn parse_event_file(event_file: &str) -> Vec<serde_json::Value> {
+    let content = fs::read(event_file).unwrap();
+
+    let mut ret = Vec::new();
+    let mut entry_start = 0;
+    let mut count = 0;
+    for (idx, &c) in content.iter().enumerate() {
+        if c as char == '{' {
+            count += 1;
+        } else if c as char == '}' {
+            assert!(count > 0);
+            count -= 1;
+        }
+
+        if count == 0 {
+            let entry = String::from_utf8_lossy(&content[entry_start..idx + 1]);
+            ret.push(serde_json::from_str(&entry).unwrap());
+            entry_start = idx + 1;
+        }
+    }
+
+    ret
+}
+
+fn check_sequential_events(expected_events: &[&MetaEvent], event_file: &str) -> bool {
```

Comment:
It'd be worth adding some comments about `check_sequential_events()`, `check_sequential_events_exact()` and `check_latest_events_exact()` to clarify what they check and the value they return.